syntax = "proto3";
package shop;

message Empty {}

message Id { int32 id = 1; }

message ItemCreate { string name = 1; double price = 2; }
message Item { int32 id = 1; string name = 2; double price = 3; bool deleted = 4; }

message CartItem { int32 id = 1; int32 quantity = 2; }
message Cart { int32 id = 1; repeated CartItem items = 2; double price = 3; }

message AddToCartRequest { int32 cart_id = 1; int32 item_id = 2; }

service Shop {
  rpc CreateCart(Empty) returns (Id);
  rpc GetCart(Id) returns (Cart);
  rpc AddToCart(AddToCartRequest) returns (Cart);
  rpc CreateItem(ItemCreate) returns (Item);
  rpc GetItem(Id) returns (Item);
}